test/functional/3-subcollections.int-spec.ts

```diff
@@ -14,7 +8,7 @@ describe('Integration test: SubCollections', () => {
   @Collection(getUniqueColName('band-with-subcollections'))
   class FullBand extends BandEntity {
     @SubCollection(Album)
-    albums: ISubCollection<Album>;
+    albums: BaseFirestoreRepository<Album>;
```

Comment:
Why did you change this type? Ideally it should be a `IRepository`? I remember that I had some problems with #172  but can't recall exactly what it was

Reply:
1) `ISubCollection` interface (that is alias of `IRepository`) does not expose `createBatch()` method - https://github.com/wovalle/fireorm/blob/1ebfde9aa983a7f0068251eb11a64a583cfb8d0b/src/types.ts#L80
2) As we can see, subcollection property becomes instance of `BaseFirestoreRepository` while deserialization:
https://github.com/wovalle/fireorm/blob/1ebfde9aa983a7f0068251eb11a64a583cfb8d0b/src/AbstractFirestoreRepository.ts#L88 and https://github.com/wovalle/fireorm/blob/1ebfde9aa983a7f0068251eb11a64a583cfb8d0b/src/helpers.ts#L13

Reply:
Thanks! @skalashnyk, makes sense. I'll have to check the typing ISubCollection type declaration, for now is ok